### ISSUE TITLE
Mistakes of Workload calculation

### ISSUE THREAD

**Issue by fly-dragon211** [bytedance/byteps#441]
The calculation of $Mss_{GPU}$ and $Mss_{CPU}$ may have some mistakes. The following is my calculation procedure:

![image](https://user-images.githubusercontent.com/55907441/223700474-7d12dde8-6333-45ba-bb89-abbba50d63dd.png)

while in the paper:
![image](https://user-images.githubusercontent.com/55907441/223701043-9b2a8802-645f-4b66-96a3-ae9336564efe.png)

**Comment by shadow150519** [comment]
i think they do make the mistake :(

**Comment by shadow150519** [comment]
> The calculation of MssGPU and MssCPU may have some mistakes. The following is my calculation procedure:
> 
> ![image](https://user-images.githubusercontent.com/55907441/223700474-7d12dde8-6333-45ba-bb89-abbba50d63dd.png)
> 
> while in the paper: ![image](https://user-images.githubusercontent.com/55907441/223701043-9b2a8802-645f-4b66-96a3-ae9336564efe.png)

Can you tell me do they build the system using the wrong result or just a slip of the pen ？

**Comment by ymjiang** [comment]
@shadow150519 Yes, this is a typo in the paper. Despite the typo, the final conclusion in equation (6) is right and BytePS correctly implements the traffic allocation strategy. Thank you for pointing out this.

**Comment by shadow150519** [comment]
@ymjiang 
base on `tc = tg` and eq3, eq4, eq5, i get the following result

$$ M_{ss_{GPU}} = \frac{k - 1}{-kn + 2k -n}*M $$

$$M_{ss_{CPU}} = \frac{2(1-n)}{-kn+2k-n}*M$$

and the t<sub>opt</sub> should be the following result

$$  t_{opt} = \frac{2(1-n)}{(-kn+2k-n)*B}*M  $$

eq7 is wrong and should be the following

$$ \gamma_{\alpha} = \frac{\frac{2(n-1)M}{nB}}{t_{opt}} = \frac{kn-2k + n}{n} $$

$$ \gamma_{p} = \frac{\frac{nM}{kB}}{t_opt}  = \frac{n(-kn+2k-n)}{2k(1-n)}$$

i think these should be the correct result

**Comment by shadow150519** [comment]
i think you do the calculation totally wrong in section 4.1
